### ISSUE TITLE
Allow sends and receives to be cancelled

### ISSUE THREAD

**Issue by Jacalz** [Jacalz/wormhole-gui#33]
### Is your feature request related to a problem? Please describe:

Sends and receives are currently not cancellable. The support for this did land upstream in [wormhole-william](https://github.com/psanford/wormhole-william) a release or two back but there are still a few issues upstream that ideally should be resolved before I feel comfortable implementing this.

### Describe the solution you'd like to see:

The work from https://github.com/Jacalz/wormhole-gui/tree/timeout should be added in (and rebased against the latest release) once https://github.com/psanford/wormhole-william/issues/53 has been resolved.

**Comment by vRobM** [comment]
Yes, sometime a receive code is typo'd OR correct but using a different relay (like croc), so it will never complete. Timeouts and cleanup is necessary.

**Comment by Jacalz** [comment]
I think I might want to include this as an experimental feature given that there might still be ways in which it doesn't quite work as expected. Initial plan is to have this in v3.7.0 (the release series after the upcoming one).
